### PR TITLE
sink: clean up useless WaitPrepare call

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve? 

part of https://github.com/pingcap/ticdc/issues/2632

### What is changed and how it works?
clean up useless WaitPrepare call.
The mounter will ensure the row is ready to use.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

None

Side effects

None

Related changes

 - Need to cherry-pick to the release branch
 

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by overvenus — APPROVED** [review]
LGTM

File: cdc/processor/pipeline/sink.go [hunk 1]
```diff
@@ -153,6 +153,8 @@ func (n *sinkNode) emitEvent(ctx pipeline.NodeContext, event *model.PolymorphicE
 		return nil
 	}
 
+	event.Row.ReplicaID = event.ReplicaID
```
[the remark this answers is not in the document]
Comment (hi-rustin) [reply cdc/processor/pipeline/sink.go]: It seems we already set it here:https://github.com/pingcap/ticdc/blob/9b50616fefbc8ae5c9efd5790e308dba6e363776/cdc/processor/pipeline/cyclic_mark.go#L159

[the remark this answers is not in the document]
Comment (hi-rustin) [reply cdc/processor/pipeline/sink.go:156]: It seems that we do need to set the ReplicaID here and the row will be incorrect after deletion. I will continue to see if there is a suitable location to set it.

File: cdc/processor/pipeline/cyclic_mark.go [hunk 2]
```diff
@@ -77,87 +79,81 @@ func (n *cyclicMarkNode) Receive(ctx pipeline.NodeContext) error {
...
+// and send events to the next node.
+func (n *cyclicMarkNode) sendNormalRowEventToNextNode(ctx pipeline.NodeContext, replicaID uint64, events ...*model.PolymorphicEvent) {
 	if _, shouldFilter := n.filterReplicaID[replicaID]; shouldFilter {
 		return
 	}
-	for _, row := range rows {
-		row.ReplicaID = replicaID
-		ctx.SendToNextNode(pipeline.PolymorphicEventMessage(row))
+	for _, event := range events {
+		event.ReplicaID = replicaID
```
Comment (amyangfei) [inline cdc/processor/pipeline/cyclic_mark.go:154]: Is `event.ReplicaID` useful after this PR

File: cdc/processor/pipeline/cyclic_mark.go [hunk 3]
```diff
@@ -56,12 +57,13 @@ func (n *cyclicMarkNode) Init(ctx pipeline.NodeContext) error {
...
 
-// Receive receives the message from the previous node
+// Receive receives the message from the previous node.
 // In the previous nodes(puller node and sorter node),
-// the change logs of mark table and normal table are listen by one puller,
-// and sorted by one sorter. So, this node will receive a commitTs-ordered stream which include the mark rows and normal rows.
-// Under the above conditions, we need to cache at most one transaction's rows to matching rows.
-// For every row event, Receive function flushes every the last transaction's rows, and adds the mark row or normal row into the cache.
+// the change logs of mark table and normal table are listen by one puller, and sorted by one sorter.
+// So, this node will receive a commitTs-ordered stream which include the mark row events and normal row events.
```
Comment (overvenus) [inline cdc/processor/pipeline/cyclic_mark.go:63]: Could you wrap line length within 80 chars?

**Review by overvenus — APPROVED** [review]
LGTM

File: cdc/processor/pipeline/cyclic_mark.go (hunk 2 above)
Comment (hi-rustin) [reply cdc/processor/pipeline/cyclic_mark.go:154]: Good catch! Let me try to remove it.

File: cdc/processor/pipeline/cyclic_mark.go (hunk 3 above)
Comment (hi-rustin) [reply cdc/processor/pipeline/cyclic_mark.go:63]: Maybe [lll](https://github.com/golangci/golangci-lint/blob/master/pkg/golinters/lll.go) will help. Let me fix it manually first.

Comment (hi-rustin) [reply cdc/processor/pipeline/cyclic_mark.go]: Done.

File: cdc/processor/pipeline/cyclic_mark.go (hunk 2 above)
Comment (hi-rustin) [reply cdc/processor/pipeline/cyclic_mark.go]: Done.

**Comment by hi-rustin** [comment]
CI failed becasue:

```log
[2021-09-01T07:07:56.194Z] grep -vE ".*.pb.go|github.com/pingcap/ticdc/testing_utils/.*|github.com/pingcap/ticdc/cdc/kv/testing.go|github.com/pingcap/ticdc/cdc/sink/simple_mysql_tester.go|.*.__failpoint_binding__.go" "/tmp/tidb_cdc_test/cov.unit.out" > "/tmp/tidb_cdc_test/unit_cov.out"

[2021-09-01T07:07:56.194Z] GO111MODULE=off go get github.com/mattn/goveralls

[2021-09-01T07:13:18.239Z] Post "https://coveralls.io/api/v1/jobs": read tcp 10.233.79.31:45000->172.67.72.111:443: read: connection timed out

[2021-09-01T07:13:18.239Z] make: *** [coverage] Error 1

```

**Review by asddongmen — APPROVED** [review]

**Pull request merged**